src/graphics/systems/mesh.rs

```diff
@@ -204,64 +204,57 @@ pub fn create() -> Box<dyn Schedulable> {
                     });
 
                     // Collect materials in to their groups.
-                    let asset_materials = asset_manager.get_materials();
-                    let pbr_materials: Vec<_> = asset_materials.iter().filter(|material| match material { Material::PBR(_) => { true }, _ => { false }}).collect();
-                    let unlit_materials: Vec<_> = asset_materials.iter().filter(|material| match material { Material::Unlit(_) => { true }, _ => { false }}).collect();
+                    let asset_materials = (asset_manager as &mut AssetManager).get_loaded_materials();
```

Comment:
Why `(asset_manager as &mut AssetManager)`? 😕

Reply:
Looking at the code I don't think we need a mutable reference to asset manager at all. 🤔

Reply:
Ups. Sometimes my rust analyzer doesn't lint so I trick it like this. I just forgot to remove it. But you are right. It shouldnt be mutable!

Reply:
Don't forget to change the system builder from a `write_resource` to a `read_resource` for the asset manager! 😄